### ISSUE TITLE
Story: Add External Link Icon to usa-prose

### ISSUE THREAD

**Issue by andyvanavery31** [NCIOCPL/ncids#1950]
### Specifications

**Visual**
- The external link icon will look the same on the listed components.
- The icon will scale to the relative size of the font-size for which it is linked.

**Functional**
- If the link text does not have .gov it will get an external link icon
- A link can be overridden to have an external link icon
- A link with an external link icon can be overridden to remove it
- The link will translate appropriately into Spanish
- Call the external link mixin

**Impacted Components**

- usa-prose anchors that do not have style applied
  - Doc Site
  - Storybook
- [ ] Summary Box (a summary box does not have links, but can contain other components that can style the links)
  - [ ] [Doc Site](https://designsystem-dev.cancer.gov/develop/components/summary-box)
  - [ ] [Storybook Summary Box Default](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-summary-box-default--default)

### DoR Tasks

- [x] Comps Created
- [ ] Comps Reviewed using [Checklist](https://github.com/NCIOCPL/cgov-digital-platform/wiki/Product-Review-of-Design-Solutions)
  - [ ] Developer Review
  - [ ] IA Review
  - [ ] Product Review
- [ ] Comps Approved
  - [ ] Developer Approval
  - [ ] IA Approval
  - [ ] Product Approval
- [ ] Scenarios Created
- [ ] Scenarios Reviewed 
  - [ ] Designer Review
  - [ ] Developer Review
  - [ ] IA Review
- [ ] Scenarios Approved
  - [ ] Designer Approval
  - [ ] Developer Approval
  - [ ] IA Approval
- [ ] Acceptance Criteria Drafted
- [ ] Acceptance Criteria Reviewed
  - [ ] Developer Review
  - [ ] IA Review
- [ ] Acceptance Criteria Approved
  - [ ] Developer Approval
  - [ ] IA Approval

### DoD Tasks

- [ ] Governance Drafted
- [ ] PR Review (see PR for review and comments)
- [ ] Client review with demo

### Technical Notes

- Determine within usa-prose, how do we target unstyled links without affecting components placed within usa-prose. (e.g., [see the card in text](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-card-test-cases--card-in-text-left))

**Scenarios/Comps**

- Use the comps here: [usa-prose in text link](https://www.figma.com/design/GCOniv8OTyj9OC1udKwo6S/Mini-Landing-Pages?node-id=8659-35288&m=dev)
- On the NCIDS Documentation Site: 
  - Create usa-prose page. This page goes in the Components section. Define when this should be used, how the domain settings and how this would be used. Guidance should be listed about when these should be used instead of how USWDS explains how to use it.
- On Storybook:
  - Add stories to the Link story (created in #1902) by adding a folder called "Test Cases". In that folder, add 3 stories:
    - 1st Story: Display anchors with no styling applied. Show the three cases: 
      - A normal external link, going to www.google.com, with the external link icon displayed.
      - An external link, going to www.google.com, with the override to not display the external link icon.
      - An internal link, going to www.cancer.gov, with the override to display the external link icon.
    - 2nd Story: Display an NCIDS Summary Box with the three anchor cases:
      - A normal external link, going to www.google.com, with the external link icon displayed
      - An external link, going to www.google.com, with the override to not display the external link icon
      - An internal link, going to www.cancer.gov, with the override to display the external link icon.

### Acceptance Criteria

```gherkin
Scenario: A link with an external url (does not contain ".gov") is viewed on the NCIDS where prose is used
  Given the external link icon has been added to the NCIDS Documentation Site and NCIDS Storybook for anchors in prose
    And the link has a url that does not contain `.gov`
  When an anchor in prose has a url that does not contain `.gov`
  Then the anchor will appear with the external link icon
    And the icon on the on the anchor will match the comp (see behavior on link from #1902)
    And the external link icon will remain center aligned vertically on the line of text it is on
  When a user wants to override the behavior of an anchor that does not contain `.gov` to not display the external link icon
  Then they will be able to override the behavior
    And the external link icon will not be displayed
  When a user has an anchor with a url that contains `.gov`
    And they want to override the behavior to include the external link icon
  Then they will be able to override the behavior
    And the external link icon will be displayed
```
